client/repo.c

```diff
@@ -489,18 +493,23 @@ TDNFGetRepoMD(
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
-    dwError = TDNFAllocateStringPrintf(&pszRepoMDFile,
-                                      "%s/%s",
-                                      pszRepoDataDir,
-                                      TDNF_REPO_METADATA_FILE_NAME);
-    BAIL_ON_TDNF_ERROR(dwError);
+    if(pRepoData->pszMetaLink)
+        metalink = 1;
 
-    dwError = TDNFAllocateStringPrintf(&pszRepoMDUrl,
-                                      "%s/%s",
-                                      pRepoData->pszBaseUrl,
-                                      TDNF_REPO_METADATA_FILE_PATH);
+    dwError = TDNFAllocateStringPrintf(&pszRepoMDFile,
+                                       "%s/%s",
+                                       pszRepoDataDir,
+                                       TDNF_REPO_METADATA_FILE_NAME);
     BAIL_ON_TDNF_ERROR(dwError);
-
+    if(metalink)
+    {
+        dwError = TDNFAllocateStringPrintf(&pszMetaLinkFile,
```

Comment:
TDNF_SAFE_FREE_MEMORY(pszMetaLinkFile) is missed in the cleanup.

Reply:
Yeah, Thanks!